### PR TITLE
Quit autotype dialog with esc key

### LINKED ISSUE #907: #improvement Make ESC work as "cancel" in the auto-type selection window
The ESC key does not work in the Auto-Type selection window.
(cursor lets you select an entry, Enter lets you use it for auto-type, Esc should cancel, no ?)

Why is that important to me ?

I actually often need to "look something up" using the auto-type shortcut, but then don't actually want to auto-type, so a simple way to escape the auto-type selection windows would be great :))

## Expected Behavior
ESC should cancel

## Current Behavior
ESC does nothing

## Debug Info
KeePassXC - Version 2.2.0
Revision: caa49a8ef3ee28ed478192389b21d61107b3b8e0

Libraries:
- Qt 5.9.1
- libgcrypt 1.8.0

Operating system: Arch Linux
CPU architecture: x86_64
Kernel: linux 4.12.8-2-ARCH

Enabled extensions:
- KeePassHTTP
- Auto-Type
- YubiKey

### PR DESCRIPTION
**Pull request by TheZ3ro**
## Description

Quit autotype dialog with esc key

## Motivation and context

Fixes #907

## Types of changes

- ✅ New feature (non-breaking change which adds functionality)

## Checklist:

- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**

### CODE REVIEW

File: src/autotype/AutoTypeSelectView.cpp [hunk 1]
```diff
@@ -54,3 +56,18 @@ void AutoTypeSelectView::selectFirstEntry()
         setCurrentIndex(index);
     }
 }
+
+void AutoTypeSelectView::keyReleaseEvent(QKeyEvent* e)
+{
+    qDebug() << e->key();
+
+    if (!e->modifiers() || (e->modifiers() & Qt::KeypadModifier && e->key() == Qt::Key_Enter)) {
+        if (e->key() == Qt::Key_Escape) {
```
[the remark this answers is not in the document]
Comment (TheZ3ro) [reply src/autotype/AutoTypeSelectView.cpp]: I copied this code from the PasswordGeneratorWidget but actually the first if seems useful in both cases.
I will fix them

[the remark this answers is not in the document]
Comment (phoerious) [reply src/autotype/AutoTypeSelectView.cpp]: What seems useful? The check for KeyPadModifier and Enter? It does absolutely nothing.

[the remark this answers is not in the document]
Comment (TheZ3ro) [reply src/autotype/AutoTypeSelectView.cpp]: lol, I was meaning useless

File: src/gui/entry/EntryView.cpp [hunk 2]
```diff
@@ -121,6 +121,20 @@ void EntryView::keyPressEvent(QKeyEvent* event)
 #endif
     }
 
+    int last = m_model->rowCount()-1;
```
Comment (phoerious) [inline src/gui/entry/EntryView.cpp]: Spaces please. ;-)

Same for if below.

Comment (TheZ3ro) [reply src/gui/entry/EntryView.cpp]: Fixed with a rebase

**Pull request merged**